components/long-click.js

```diff
@@ -0,0 +1,99 @@
+// node_modules
+import { useEffect, useRef } from "react";
+
+/**
+ * Default amount of time in milliseconds to wait before considering a mouse click a long press.
+ */
+const DEFAULT_DELAY = 250;
+
+/**
+ * Hook to detect short and long clicks in an element. Pass it the ID of the element to attach the
+ * event listeners to. Pass it the regular- and long-click callbacks. It calls the regular-click
+ * callback when the user short-clicks the element, and the long-click callback when the user
+ * long-clicks the element.
+ *
+ * It returns an object with a function to call to trigger the mouse-up action, if some other way
+ * of triggering the mouse-up action is needed. As an example, a checkbox needs to handle clicks
+ * itself for accessibility, so its `onClick` handler calls the `onClickEnd` function returned by
+ * this hook.
+ *
+ * This hook also returns the ID of the long-click timer, in case it needs clearing due to some
+ * event that happens outside this hook.
+ * @param {string} elementId Unique ID of the element to attach the event listeners to
+ * @param {function} onClick Called when the user short-clicks the element
+ * @param {function} onLongClick Called when the user long-clicks the element
+ * @param {number} delay? Time in milliseconds to wait before considering a mouse click a long press
+ * @returns {object} Contains the `onClickEnd` callback and the `timer` object
+ */
+export default function useLongClick(
+  elementId,
+  onClick,
+  onLongClick,
+  delay = DEFAULT_DELAY
+) {
+  // The "UNSET" timer value indicates that the timer isn't active. That's as opposed to the null
+  // value, which indicates that the timer has expired
+  const timer = useRef("UNSET");
```

Comment:
in typescript land I think we could use an enum or something to express this. Such is the js life though for now 🤷